### DISCUSSION TITLE
Can't use didFinishUpdateCycleForUpdateCheck

### DISCUSSION THREAD

**Discussion by nmilev-scribe** [sparkle-project/Sparkle#2391] [Q&A] [answered]
### Description of the problem
I am integrating our MacOS app with Sparkle and everything works properly. However, I now want to introduce a callback whenever the app is updated. I found out in the docs that I can do something like this:
```

class AppDelegate: SPUUpdaterDelegate {
    ...
    var updaterController: SPUStandardUpdaterController? = nil
    ...
    func applicationDidFinishLaunching(_ aNotification: Notification) {
            updaterController = SPUStandardUpdaterController(startingUpdater: true, updaterDelegate: self, userDriverDelegate: nil)
    ...
    }
    ...
    func updater(_ updater: SPUUpdater, didFinishUpdateCycleFor updateCheck: SPUUpdateCheck, error: Error?) {
        debugPrint("Update cycle finished")
    }
...
}
```
However, this callback is never executed. What am I doing wrong?

### Do you use Sandboxing in your app?
No
### Version of `Sparkle.framework` in the latest version of your app
2.1.0
### Version of `Sparkle.framework` in the old version of app that your users have (or N/A)
2.1.0
### Sparkle's output from Console.app
```

```

### Steps to reproduce the behavior

[The more information provided and pasted verbatim, the easier it will be to diagnose an issue. If you can provide the affected application/binary and XML feed to reproduce an issue, share them]

**Comment by zorgiepoo** [comment]
It won’t be called in this case for an app updating itself and having been relaunched. Why do you want to know when an update is installed and what are you actually trying to do? What do you do when the user updates your app manually, not using Sparkle? Or if an update is installed silently without the user knowing.

**Reply by nmilev-scribe** [reply]
The signing Team is a colleague's personal team and we want to move the app to another team (that several of us can use, therefore being able to create new releases of the app). Since we do require some permissions from the user, they will remain granted, but the app won't work (I'm guessing the signing team change messes this up). Therefore, I'd like to detect when the app has been upgraded to a certain version and just use a shell command to revoke the permissions given by the user. Then, they will be asked to give them again.

I suppose I can do this another way but I thought this would be the easiest way.

**Reply by zorgiepoo** [reply, marked as answer]
Since the user can update your app independently from Sparkle I suggest doing it another way. This is one reason why we haven’t exposed an API for when an app updates/relaunches itself. It becomes a double edge sword when it should not be used in cases like this.

**Reply by nmilev-scribe** [reply]
Okay, thanks for clarifying!
